staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -820,6 +820,34 @@ type CephFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty" protobuf:"varint,6,opt,name=readOnly"`
 }
 
+// Represents a Ceph Filesystem mount that lasts the lifetime of a pod
+// Cephfs volumes do not support ownership management or SELinux relabeling.
+type CephFSPersistentVolumeSource struct {
+	// Required: Monitors is a collection of Ceph monitors
+	// More info: https://releases.k8s.io/HEAD/examples/volumes/cephfs/README.md#how-to-use-it
+	Monitors []string `json:"monitors" protobuf:"bytes,1,rep,name=monitors"`
+	// Optional: Used as the mounted root, rather than the full Ceph tree, default is /
+	// +optional
+	Path string `json:"path,omitempty" protobuf:"bytes,2,opt,name=path"`
+	// Optional: User is the rados user name, default is admin
+	// More info: https://releases.k8s.io/HEAD/examples/volumes/cephfs/README.md#how-to-use-it
+	// +optional
+	User string `json:"user,omitempty" protobuf:"bytes,3,opt,name=user"`
+	// Optional: SecretFile is the path to key ring for User, default is /etc/ceph/user.secret
+	// More info: https://releases.k8s.io/HEAD/examples/volumes/cephfs/README.md#how-to-use-it
+	// +optional
+	SecretFile string `json:"secretFile,omitempty" protobuf:"bytes,4,opt,name=secretFile"`
+	// Optional: SecretRef is reference to the authentication secret for User, default is empty.
+	// More info: https://releases.k8s.io/HEAD/examples/volumes/cephfs/README.md#how-to-use-it
+	// +optional
+	SecretRef *ObjectReference `json:"secretRef,omitempty" protobuf:"bytes,5,opt,name=secretRef"`
```

Comment:
@liggitt @jsafrane what about adding a `SecretNamespace` object? Then we don't have to worry about conversion issue.

```go
type CephFSVolumeSource struct {
    // Required: Monitors is a collection of Ceph monitors                                                                                                                            
    Monitors []string
    // Optional: Used as the mounted root, rather than the full Ceph tree, default is /                                                                                               
    // +optional                                                                                                                                                                      
    Path string
    // Optional: User is the rados user name, default is admin                                                                                                                        
    // +optional                                                                                                                                                                      
    User string
    // Optional: SecretFile is the path to key ring for User, default is /etc/ceph/user.secret                                                                                        
    // +optional                                                                                                                                                                      
    SecretFile string
    // Optional: SecretRef is reference to the authentication secret for User, default is empty.                                                                                      
    // +optional                                                                                                                                                                      
    SecretRef *LocalObjectReference
    // Optional: Defaults to false (read/write). ReadOnly here will force                                                                                                             
    // the ReadOnly setting in VolumeMounts.                                                                                                                                          
    // +optional                                                                                                                                                                      
    ReadOnly bool
    // Optional: Namespace for SecretRef, default is empty.                                                                                      
    // +optional                                                                                                                                                                      
    SecretNamespace *string
}